cfme/common/vm_views.py

```diff
@@ -77,33 +80,36 @@ class JSInstanceEntity(JSBaseEntity):
     @property
     def data(self):
         data_dict = super(JSInstanceEntity, self).data
-        if 'quadicon' in data_dict and data_dict['quadicon']:
-            quad_data = document_fromstring(data_dict['quadicon'])
-            data_dict['os'] = quad_data.xpath(self.QUADRANT.format(pos="a"))[0].get('src')
-            data_dict['vendor'] = quad_data.xpath(self.QUADRANT.format(pos="c"))[0].get('src')
-            data_dict['no_snapshot'] = quad_data.xpath(self.QUADRANT.format(pos="d"))[0].text
+        try:
```

Comment:
**QUESTION**
I don't understand why `try` block is required here? Where `IndexError` can be raised?

Reply:
@quarckster 
When there will be no 'os', 'vendor' and 'no_snapshot' index present in data_dict then IndexError will be raised.